pennylane/devices/default_qubit_jax.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains an jax implementation of the :class:`~.DefaultQubit`
+reference plugin.
+"""
+
+
+from pennylane.operation import DiagonalOperation
+from pennylane.devices import DefaultQubit
+from pennylane.devices import jax_ops
+import numpy as np
+
+try:
+    import jax.numpy as jnp
+    import jax
+
+except ImportError as e:  # pragma: no cover
+    raise ImportError("default.qubit.jax device requires installing jax>0.2.0") from e
+
+
+class DefaultQubitJax(DefaultQubit):
+    """Simulator plugin based on ``"default.qubit"``, written using jax.
+
+    **Short name:** ``default.qubit.jax``
+
+    This device provides a pure-state qubit simulator written using jax. As a result, it
+    supports classical backpropagation as a means to compute the gradient. This can be faster than
+    the parameter-shift rule for analytic quantum gradients when the number of parameters to be
+    optimized is large.
+
+    To use this device, you will need to install jax:
+
+    .. code-block:: console
+
+        pip install jax jaxlib
+
+    **Example**
+
+    The ``default.qubit.jax`` is designed to be used with end-to-end classical backpropagation
+    (``diff_method="backprop"``) with the jax interface. This is the default method of
+    differentiation when creating a QNode with this device.
+
+    Using this method, the created QNode is a 'white-box', and is
+    tightly integrated with your jax computation:
+
+    >>> dev = qml.device("default.qubit.jax", wires=1)
+    >>> @qml.qnode(dev, interface="jax", diff_method="backprop")
+    ... def circuit(x):
+    ...     qml.RX(x[1], wires=0)
+    ...     qml.Rot(x[0], x[1], x[2], wires=0)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> weights = jnp.array([0.2, 0.5, 0.1])
+    >>> # Squeeze is needed since JAX requires scalar outputs for grad.
+    >>> grad_fn = jax.grad(lambda x: circuit(x).squeeze())
+    >>> print(grad_fn(weights))
+    array([-2.2526717e-01 -1.0086454e+00  1.3877788e-17])
+
+    There are a couple of things to keep in mind when using the ``"backprop"``
+    differentiation method for QNodes:
+
+    * You must use the ``"jax"`` interface for classical backpropagation, as jax is
+      used as the device backend.
+
+    .. UsageDetails::
+
+        Jax does randomness is a special way when compared to numpy, in that all PRNGs need to

```

Comment:
I *think* we might use "NumPy" when written as text. Might be good to do a quick search over the code if there is any implicit convention